.ci/gdbstub-test.sh

```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+set -e -u -o pipefail
```

Comment:
@jserv Looks like this will cause the CI to fail. 

The previous successful runs have the following log
```
...
Submodule 'mini-gdbstub' (https://github.com/RinHizakura/mini-gdbstub) registered for path 'src/mini-gdbstub'
Cloning into '/home/runner/work/rv32emu/rv32emu/src/mini-gdbstub'...
Submodule path 'src/mini-gdbstub': checked out '0d8f3c10e05c1e566cc4db67af50a55ed03c7321'
make -C src/mini-gdbstub/ O=/home/runner/work/rv32emu/rv32emu/build/mini-gdbstub/
make[1]: Entering directory '/home/runner/work/rv32emu/rv32emu/src/mini-gdbstub'
cc -c -Iinclude -Wall -Wextra -MMD  -O3 ./lib/gdbstub.c -o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//gdbstub.o
cc -c -Iinclude -Wall -Wextra -MMD  -O3 ./lib/conn.c -o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//conn.o
cc -c -Iinclude -Wall -Wextra -MMD  -O3 ./lib/utils/csum.c -o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//csum.o
cc -c -Iinclude -Wall -Wextra -MMD  -O3 ./lib/utils/translate.c -o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//translate.o
cc -c -Iinclude -Wall -Wextra -MMD  -O3 ./lib/packet.c -o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//packet.o
ar -rcs /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//libgdbstub.a /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//gdbstub.o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//conn.o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//csum.o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//translate.o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//packet.o
make[1]: Leaving directory '/home/runner/work/rv32emu/rv32emu/src/mini-gdbstub'
  CC	build/decode.o
  CC	build/io.o
  CC	build/syscall.o
  CC	build/emulate.o
  CC	build/riscv.o
  CC	build/elf.o
  CC	build/cache.o
  CC	build/mpool.o
  CC	build/syscall_sdl.o
  CC	build/gdbstub.o
  CC	build/breakpoint.o
  CC	build/main.o
  LD	build/rv32emu
lto-wrapper: warning: using serial compilation of 2 LTRANS jobs
Remote connection closed
success in [20](https://github.com/sysprog21/rv32emu/actions/runs/6794826800/job/18472689725#step:7:21)05 trials
inferior exit code 0
```

With this check, the run will fail with log 
```
...
git submodule update --init src/mini-gdbstub/
Submodule 'mini-gdbstub' (https://github.com/RinHizakura/mini-gdbstub) registered for path 'src/mini-gdbstub'
Cloning into '/home/runner/work/rv32emu/rv32emu/src/mini-gdbstub'...
Submodule path 'src/mini-gdbstub': checked out '0d8f3c10e05c1e566cc4db67af50a55ed03c7321'
make -C src/mini-gdbstub/ O=/home/runner/work/rv32emu/rv32emu/build/mini-gdbstub/
make[1]: Entering directory '/home/runner/work/rv32emu/rv32emu/src/mini-gdbstub'
cc -c -Iinclude -Wall -Wextra -MMD  -O3 ./lib/gdbstub.c -o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//gdbstub.o
cc -c -Iinclude -Wall -Wextra -MMD  -O3 ./lib/conn.c -o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//conn.o
cc -c -Iinclude -Wall -Wextra -MMD  -O3 ./lib/utils/csum.c -o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//csum.o
cc -c -Iinclude -Wall -Wextra -MMD  -O3 ./lib/utils/translate.c -o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//translate.o
cc -c -Iinclude -Wall -Wextra -MMD  -O3 ./lib/packet.c -o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//packet.o
ar -rcs /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//libgdbstub.a /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//gdbstub.o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//conn.o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//csum.o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//translate.o /home/runner/work/rv32emu/rv32emu/build/mini-gdbstub//packet.o
make[1]: Leaving directory '/home/runner/work/rv32emu/rv32emu/src/mini-gdbstub'
  CC	build/decode.o
  CC	build/io.o
  CC	build/syscall.o
  CC	build/emulate.o
  CC	build/riscv.o
  CC	build/elf.o
  CC	build/cache.o
  CC	build/mpool.o
  CC	build/syscall_sdl.o
  CC	build/gdbstub.o
  CC	build/breakpoint.o
  CC	build/main.o
  LD	build/rv[32](https://github.com/sysprog21/rv32emu/actions/runs/6794826800/job/18472689725#step:7:33)emu
lto-wrapper: warning: using serial compilation of 2 LTRANS jobs
make: *** [Makefile:110: gdbstub-test] Error 2
Error: Process completed with exit code 2.
```

Reply:
Alternatively, should you refine the return code handling?